core/src/main/java/org/elasticsearch/transport/ConnectionProfile.java

```diff
@@ -38,13 +39,22 @@
      * A pre-built light connection profile that shares a single connection across all
      * types.
      */
-    public static final ConnectionProfile LIGHT_PROFILE = new ConnectionProfile(
-        Collections.singletonList(new ConnectionTypeHandle(0, 1, EnumSet.of(
-            TransportRequestOptions.Type.BULK,
-            TransportRequestOptions.Type.PING,
-            TransportRequestOptions.Type.RECOVERY,
-            TransportRequestOptions.Type.REG,
-            TransportRequestOptions.Type.STATE))), 1, null, null);
+    public static final ConnectionProfile LIGHT_PROFILE = getLightProfileWithTimeout(null, null);
+
+    /**
+     * Builds a light connection profile that shares a single connection across all
+     * types, using the given timeouts
+     */
+    public static ConnectionProfile getLightProfileWithTimeout(@Nullable TimeValue connectTimeout,
+                                                               @Nullable TimeValue handshakeTimeout) {
+        return new ConnectionProfile(
```

Comment:
I tried to implement your suggestion and I think it looks good. will push shortly.